src/utils/utils-vega.js

```diff
@@ -709,7 +721,35 @@ export function getScales(
     })
   }
 
-  if (typeof color === "object" && color.type === "density") {
+  if (
+    typeof color === "object" &&
+    color.type === "density" &&
+    parseColorRamps(colorRamps).length === color.range.length - 1
+  ) {
+    scales.push({
+      name: getColorScaleName(layerName),
+      type: "threshold",
+      domain: parseColorRamps(colorRamps),
+      range: color.range
+        .map(c => adjustOpacity(c, color.opacity))
+        .map((c, i, colorArray) => {
+          const normVal = i / (colorArray.length - 1)
+          let interp = Math.min(normVal / 0.65, 1.0)
+          interp = interp * 0.375 + 0.625
+          return adjustRGBAOpacity(c, interp)
+        }),
+      accumulator: "density",
+      minDensityCnt: "-2ndStdDev",
+      maxDensityCnt: "2ndStdDev",
+      clamp: true
+    })
+  }
+
+  if (
```

Comment:
Might be more clear to write this something like:

```js
if (typeof color  === "object" && color.type === "density") {
  const parsedColorRamps = parseColorRamps(colorRamps)
  const domain = parsedColorRamps.length === color.range.length - 1
    ? parsedColorRamps
    : // original domain code
  // original code, but using the domain variable
}
```

Saves you from parsing the color ramps multiple times and duplicating a bunch of code.

Reply:
Fixed - great catch